src/components/FilterBar/FilterBar.test.tsx

```diff
@@ -138,15 +150,66 @@ const MulitSelectTestGroup = (multiSelectItems) => (
   </MultiSelectGroup>
 );
 describe("FilterBar Accessibility", () => {
-  it("should have no axe violations", async () => {
+  beforeAll(() => {
+    matchMedia = new MatchMedia();
+    window.resizeTo = function resizeTo(width, height) {
+      Object.assign(this, {
+        innerWidth: width,
+        innerHeight: height,
+        outerWidth: width,
+        outerHeight: height,
+      }).dispatchEvent(new this.Event("resize"));
+    };
+  });
+  afterEach(() => {
+    matchMedia.clear();
+  });
+  it("should have no axe violations on desktop", async () => {
+    window.resizeTo(1024, 600);
     const { container } = render(<FilterBarTestComponent />);
+    // Make sure it renders desktop components
+    expect(screen.getByRole("button", { name: /colors/i })).toBeInTheDocument();
+    expect(await axe(container)).toHaveNoViolations();
+  });
+  it("should have no axe violations on mobile", async () => {
+    window.resizeTo(300, 300);
+    const { container } = render(<FilterBarTestComponent />);
+    // Make sure it renders mobile components
+    expect(
+      screen.getByRole("button", { name: /show filter/i })
+    ).toBeInTheDocument();
+
     expect(await axe(container)).toHaveNoViolations();
   });
 });
+
 describe("FilterBar", () => {
-  it("should clear all selectedItems when 'Clear All' button is clicked", () => {
-    render(<FilterBarTestComponent />);
-    userEvent.click(screen.getByRole("button", { name: /show filter/i }));
+  beforeAll(() => {
+    matchMedia = new MatchMedia();
+    window.resizeTo = function resizeTo(width, height) {
+      Object.assign(this, {
+        innerWidth: width,
+        innerHeight: height,
+        outerWidth: width,
+        outerHeight: height,
+      }).dispatchEvent(new this.Event("resize"));
+    };
+  });
+  afterEach(() => {
+    matchMedia.clear();
+  });
+  it("should render a 'Clear Filters' button when showClearAll prop is passed", () => {
+    window.resizeTo(1024, 600);
+    render(<FilterBarTestComponent showClearAll id="filter-bar-test-1" />);
+    expect(
+      screen.getByRole("button", { name: /clear filters/i })
+    ).toBeInTheDocument();
+  });
+  it("should clear the selectedItems when 'Clear Filters' button is clicked", () => {
+    const { rerender } = render(
+      <FilterBarTestComponent showClearAll id="filter-bar-test-2" />
+    );
+    rerender(<FilterBarTestComponent showClearAll id="filter-bar-test-2" />);
```

Comment:
Yes so that's where I could not find a better solution. If you  add `console.log(id, isLargerThanMedium)` in FilterBar.tsx you will see that for each test the isLargerThanMedium is returning false before the rerender (exept for the first `it()` block. I tried `await`, `await waitFor()` without success, I tried to call window resize in different places and played around with the setup but this was the only thing I was able to get to work. If you have a hunch I'd love to know where to look further.